### DISCUSSION TITLE
How to set nodeOrigin per single Node without effect other Nodes

### DISCUSSION THREAD

**Discussion by sudarsanarao995** [xyflow/xyflow#3641] [General]
Please Provide one example for implement  of NodeOrigin only in  single Node, it should not effects the other nodes

**Comment by bcakmakoglu** [comment]
As per the docs, node origin can be passed per node, here's an excerpt of the [`Node`](https://reactflow.dev/api-reference/types/node) type:

```ts
export type Node<T, U extends string> = {
  // ...
  
  // set it on a per-node basis 
  origin?: NodeOrigin;
  
  // ...
};
```

So if you want it only for a single node, just use `node.origin` :)

Example:
```tsx
const initialNodes: Node[] = [
  {
    id: "1",
    type: "input",
    data: { label: "Node 1" },
	origin: [0.5, 0.5],
    position: { x: 250, y: 5 }
  },
  { id: "2", data: { label: "Node 2" }, position: { x: 100, y: 100 } },
  { id: "3", data: { label: "Node 3" }, position: { x: 400, y: 100 } },
  {
    id: "4",
    type: "custom",
    data: { label: "Custom Node" },
    position: { x: 400, y: 200 }
  }
];
```

**Reply by moklick** [reply]
Sorry, this was a confusion with React Flow v12. You need to wait for the next major release in order to use this attribute.

**Reply by sudarsanarao995** [reply]
we are using in React Flow version 11. please confirm me , is it possible  to implement nodeOrigin for only singleNode without effects another nodes.

**Reply by bcakmakoglu** [reply]
Oh I didn't know it's not in v11 yet 🥲 My bad then 😄

**Comment by sudarsanarao995** [comment]
> As per the docs, node origin can be passed per node, here's an excerpt of the [`Node`](https://reactflow.dev/api-reference/types/node) type:
> 
> ```ts
> export type Node<T, U extends string> = {
>   // ...
>   
>   // set it on a per-node basis 
>   origin?: NodeOrigin;
>   
>   // ...
> };
> ```
> 
> So if you want it only for a single node, just use `node.origin` :)
> 
> Example:
> 
> ```tsx
> const initialNodes: Node[] = [
>   {
>     id: "1",
>     type: "input",
>     data: { label: "Node 1" },
> 	origin: [0.5, 0.5],
>     position: { x: 250, y: 5 }
>   },
>   { id: "2", data: { label: "Node 2" }, position: { x: 100, y: 100 } },
>   { id: "3", data: { label: "Node 3" }, position: { x: 400, y: 100 } },
>   {
>     id: "4",
>     type: "custom",
>     data: { label: "Custom Node" },
>     position: { x: 400, y: 200 }
>   }
> ];
> ```
@bcakmakoglu In ReactFlow version 11 in my node_Modules , I did not seen this type of  **origin?: NodeOrigin**; please observe the below code

```
export type Node<T = any, U extends string | undefined = string | undefined> = {
    id: string;
    position: XYPosition;
    data: T;
    type?: U;
    style?: CSSProperties;
    className?: string;
    sourcePosition?: Position;
    targetPosition?: Position;
    hidden?: boolean;
    selected?: boolean;
    dragging?: boolean;
    draggable?: boolean;
    selectable?: boolean;
    connectable?: boolean;
    deletable?: boolean;
    dragHandle?: string;
    width?: number | null;
    height?: number | null;
    parentNode?: string;
    zIndex?: number;
    extent?: 'parent' | CoordinateExtent;
    expandParent?: boolean;
    positionAbsolute?: XYPosition;
    ariaLabel?: string;
    focusable?: boolean;
    resizing?: boolean;
    [internalsSymbol]?: {
        z?: number;
        handleBounds?: NodeHandleBounds;
        isParent?: boolean;
    };
};
```
so it can't possible to use origin in  single Node.

**if it is possible could you please show me in React.js code , if possible please show me the code in codesandbox**
